src/lib/progress-bar/progress-bar.ts

```diff
@@ -82,6 +88,18 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
   set bufferValue(v: number) { this._bufferValue = clamp(v || 0); }
   private _bufferValue: number = 0;
 
+  @ViewChild("primaryValueBar") primaryValueBar: ElementRef;
+
+  /** 
+   * Event that indicates animation end on value bar. Currently only support determinate and buffer. 
+   * Also note, if multiple event update is thrown in quick succession the last event completion is 
+   * the one firing the output.
+   */
+  @Output() valueAnimationEnd = new EventEmitter<number>();
+
+  /** Reference to animation end subscription to be unsubscribed on destroy. */
+  private _valueAnimationSubscription: Subscription | null;
```

Comment:
You can initialize this to `Subscription.EMPTY`. That way you won't have to null check it everywhere.